File: app/src/main/java/com/veganbeauty/app/features/shop/search/SearchSuggestionAdapter.java
package com.veganbeauty.app.features.shop.search;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.databinding.ShopSearchBarBinding;

import java.util.ArrayList;
import java.util.List;

public class SearchSuggestionAdapter extends RecyclerView.Adapter<SearchSuggestionAdapter.ViewHolder> {

    public interface OnItemClickListener {
        void onItemClick(String text);
    }

    private final int iconRes;
    private final OnItemClickListener onItemClick;
    private final List<String> items = new ArrayList<>();

    public SearchSuggestionAdapter(int iconRes, OnItemClickListener onItemClick) {
        this.iconRes = iconRes;
        this.onItemClick = onItemClick;
    }

    public void submitList(List<String> newItems) {
        items.clear();
        items.addAll(newItems);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ShopSearchBarBinding binding = ShopSearchBarBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final ShopSearchBarBinding binding;

        public ViewHolder(ShopSearchBarBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(String text) {
            binding.tvText.setText(text);
            binding.ivIcon.setImageResource(iconRes);
            binding.getRoot().setOnClickListener(v -> onItemClick.onItemClick(text));
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/shop/search/SearchSuggestionAdapter.kt
package com.veganbeauty.app.features.shop.search

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.veganbeauty.app.databinding.ShopSearchBarBinding

class SearchSuggestionAdapter(
    private val iconRes: Int,
    private val onItemClick: (String) -> Unit
) : RecyclerView.Adapter<SearchSuggestionAdapter.ViewHolder>() {

    private val items = mutableListOf<String>()

    fun submitList(newItems: List<String>) {
        items.clear()
        items.addAll(newItems)
        notifyDataSetChanged()
    }

    inner class ViewHolder(private val binding: ShopSearchBarBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(text: String) {
            binding.tvText.text = text
            binding.ivIcon.setImageResource(iconRes)
            binding.root.setOnClickListener { onItemClick(text) }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ShopSearchBarBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun getItemCount() = items.size
}
